libraries/core_libs/consensus/src/dag/block_proposer.cpp

```diff
@@ -217,9 +217,6 @@ void BlockProposer::proposeBlock(DagFrontier&& frontier, level_t level, uint64_t
   for (const auto& trx : trxs) {
     auto weight = trx_mgr_->estimateTransactionGas(trx, proposal_period);
     block_weight += weight;
```

Comment:
`estimateTransactionGas` is expensive call and now you are executing it 2x. I would modify this code too, so as we already calling it from `packTrxs`. And `packTrxs` should also return estimations array